sys/linux/net_80211.txt

```diff
@@ -55,3 +64,204 @@ define JOIN_IBSS_BG_SCAN	0x1
+
+ieee80211_vht_capabilities {
+	vht_cap_info	flags[ieee80211_vht_cap_info, int32]
+	vht_mcs_info	int64
+} [packed]
+
+# As defined by Table 9-45 of IEEE 802.11-2016
+type ieee80211_reason_code[BASE_TYPE] BASE_TYPE[0:66]
+
+# As defined by Table 9-46 of IEEE 802.11-2016
+type ieee80211_status_code[BASE_TYPE] BASE_TYPE[0:107]
+
+# Only NAV is supported at the moment. No CPE or PS-Poll frames
+ieee80211_duration {
+	duration	int16:15
+	nav_flag	const[0, int16:1]
+} [packed]
+
+# As defined in sect. 9.2.4.4.1 of IEEE 802.11-216
+ieee80211_seq_control {
+	frag_number	int16:4
+	seq_number	int16:12
+} [packed]
+
+# As defined by Table 9-342 of IEEE 802.11-2016
+ieee80211_tdls_action_codes = WLAN_TDLS_SETUP_REQUEST, WLAN_TDLS_SETUP_RESPONSE, WLAN_TDLS_SETUP_CONFIRM, WLAN_TDLS_TEARDOWN, WLAN_TDLS_PEER_TRAFFIC_INDICATION, WLAN_TDLS_CHANNEL_SWITCH_REQUEST, WLAN_TDLS_CHANNEL_SWITCH_RESPONSE, WLAN_TDLS_PEER_PSM_REQUEST, WLAN_TDLS_PEER_PSM_RESPONSE, WLAN_TDLS_PEER_TRAFFIC_RESPONSE, WLAN_TDLS_DISCOVERY_REQUEST
+
+# As defined by Table 9-131 of IEEE 802.11-2016
+ieee80211_cipher_suites = WLAN_CIPHER_SUITE_USE_GROUP, WLAN_CIPHER_SUITE_WEP40, WLAN_CIPHER_SUITE_TKIP, WLAN_CIPHER_SUITE_CCMP, WLAN_CIPHER_SUITE_WEP104, WLAN_CIPHER_SUITE_AES_CMAC, WLAN_CIPHER_SUITE_GCMP, WLAN_CIPHER_SUITE_GCMP_256, WLAN_CIPHER_SUITE_CCMP_256, WLAN_CIPHER_SUITE_BIP_GMAC_128, WLAN_CIPHER_SUITE_BIP_GMAC_256, WLAN_CIPHER_SUITE_BIP_CMAC_256
+
+# As defined by Table 9-133 of IEEE 802.11-2016
+ieee80211_akm_suites = WLAN_AKM_SUITE_8021X, WLAN_AKM_SUITE_PSK, WLAN_AKM_SUITE_FT_8021X, WLAN_AKM_SUITE_FT_PSK, WLAN_AKM_SUITE_8021X_SHA256, WLAN_AKM_SUITE_PSK_SHA256, WLAN_AKM_SUITE_TDLS, WLAN_AKM_SUITE_SAE, WLAN_AKM_SUITE_FT_OVER_SAE, WLAN_AKM_SUITE_AP_PEER_KEY, WLAN_AKM_SUITE_8021X_SUITE_B, WLAN_AKM_SUITE_8021X_SUITE_B_192, WLAN_AKM_SUITE_FT_8021X_SHA384, WLAN_AKM_SUITE_FILS_SHA256, WLAN_AKM_SUITE_FILS_SHA384, WLAN_AKM_SUITE_FT_FILS_SHA256, WLAN_AKM_SUITE_FT_FILS_SHA384, WLAN_AKM_SUITE_OWE, WLAN_AKM_SUITE_FT_PSK_SHA384, WLAN_AKM_SUITE_PSK_SHA384
+
+# Capability Information field (see sect. 9.4.1.4 of IEEE 802.11-2016)
+ieee80211_capabilities = WLAN_CAPABILITY_ESS, WLAN_CAPABILITY_IBSS, WLAN_CAPABILITY_CF_POLLABLE, WLAN_CAPABILITY_CF_POLL_REQUEST, WLAN_CAPABILITY_PRIVACY, WLAN_CAPABILITY_SHORT_PREAMBLE, WLAN_CAPABILITY_PBCC, WLAN_CAPABILITY_CHANNEL_AGILITY, WLAN_CAPABILITY_SPECTRUM_MGMT, WLAN_CAPABILITY_QOS, WLAN_CAPABILITY_SHORT_SLOT_TIME, WLAN_CAPABILITY_APSD, WLAN_CAPABILITY_RADIO_MEASURE, WLAN_CAPABILITY_DSSS_OFDM, WLAN_CAPABILITY_DEL_BACK, WLAN_CAPABILITY_IMM_BACK
+
+type ieee80211_capability[TYPE] flags[ieee80211_capabilities, TYPE]
+
+#############################################
+# Basic 802.11 frame structures
+#############################################
+
+ieee80211_ht_control_80211n {
```

Comment:
Same here, which kernel file/function parses these structures?

Reply:
HT control is ignored by mac80211 (it cuts it [here](https://elixir.bootlin.com/linux/latest/source/net/mac80211/rx.c#L129)), and is described not as an int32 only to generate packets that are valid with respect to the specs.

Reply:
Does it then get interpreted anywhere in the kernel? Because if not, then there's no value in having such detailed descriptions.